### DISCUSSION TITLE
Not casting on pivot table?

### DISCUSSION THREAD

**Discussion by elfeffe** [laravel/framework#42623] [Q&A]
I use the latest Laravel and PHP 8
I have a pivot table with a meta field which is a JSON, and then I cast like an object (and I tried with array).

$variant->pivot->meta->default_value

I have an error, it says meta is a string.

**Comment by rodrigopedra** [comment]
Are using a custom pivot model?

https://laravel.com/docs/9.x/eloquent-relationships#defining-custom-intermediate-table-models

In this custom pivot model you can add the `$casts` property and specify the field should be cast to JSON.

https://laravel.com/docs/9.x/eloquent-mutators#attribute-casting
